### ISSUE TITLE
Refactor control.lua

### ISSUE THREAD

**Issue by theis999** [MortenTobiasNielsen/Factorio-GUI-V2#352]
The control code could use a look over, and perhaps some performance enhancements.

**Comment by MortenTobiasNielsen** [comment]
Agreed. It is though a complex  task to make changes to the control flow without introducing breaking changes. That could though be a good reason to look it over - it shouldn't be too difficult to make changes.

**Comment by theis999** [comment]
It has been a while since i last took a good look at it. So don't know exactly what i would change.
Staying away from breaking changes obviously has to be a priority, which mostly means staying away from walking

**Comment by MortenTobiasNielsen** [comment]
Yeah, walking, order of execution, and tick consumption. 😅
